src/script/broadcast/BroadcastRepository.ts

```diff
@@ -0,0 +1,149 @@
+import {User} from '../entity/User';
+import {BackendClientError} from '../error/BackendClientError';
+import {WebAppEvents} from '../event/WebApp';
+import {MessageSender} from '../message/MessageSender';
+import {BroadcastService} from './BroadcastService';
+
+export class BroadcastRepository {
+  private readonly broadcastService: BroadcastService;
+  private readonly clientMismatchHandler: ClientMismatchHandler;
+  private readonly clientRepository: ClientRepository;
+  private readonly conversationRepository: ConversationRepository;
+  private readonly cryptographyRepository: CryptographyRepository;
+  private readonly logger: Logger;
+  private readonly messageSender: MessageSender;
+
+  /**
+   * @param {BroadcastService} broadcastService - Backend REST API broadcast service implementation
+   * @param {ClientRepository} clientRepository - Repository for client interactions
+   * @param {ConversationRepository} conversationRepository - Repository for conversation interactions
+   * @param {CryptographyRepository} cryptographyRepository - Repository for all cryptography interactions
+   * @param {MessageSender} messageSender - Responsible for queueing and sending messages
+   */
+  constructor(
+    broadcastService: BroadcastService,
+    clientRepository: ClientRepository,
+    conversationRepository: ConversationRepository,
+    cryptographyRepository: CryptographyRepository,
+    messageSender: MessageSender,
+  ) {
+    this.broadcastService = broadcastService;
+    this.clientRepository = clientRepository;
+    this.conversationRepository = conversationRepository;
+    this.cryptographyRepository = cryptographyRepository;
+    this.messageSender = messageSender;
+    this.logger = getLogger('BroadcastRepository');
+
+    this.clientMismatchHandler = this.conversationRepository.clientMismatchHandler;
+
+    /*
+     * FIXME this should not be handled by an event. This an action we want to perform, thus should be a direct method call.
+     * To do that, we need to inject the BroadcastRepository into the UserRepository.
+     * But this will create a cyclic dependency that we need to resolve first.
+     * As of now, the cyclic dependency would go like this:
+     *   - ConversationRepo needs UserRepository
+     *   - UserRepostory needs BroadcastRepository
+     *   - BroadcastRepository needs ConversationRepository
+     *
+     * Needing the ConversationRepository in the BroadcastRepository doesn't make sense. We need to get rid of that dependency
+     * The heavy lifting resides in generalizing the `clientMismatchHandler` so that it doesn't need to directly call the ConversationRepo
+     */
+    amplify.subscribe(
+      WebAppEvents.BROADCAST.SEND_MESSAGE,
+      ({genericMessage, recipients}: {genericMessage: GenericMessage; recipients: User[]}) => {
+        this.broadcastGenericMessage(genericMessage, recipients);
+      },
+    );
+  }
+
+  /**
+   * @param genericMessage Generic message that will be send
```

Comment:
```suggestion
   * @param genericMessage - Generic message that will be send
```

Reply:
I am against this, the `-` will appear before the message when hovering over the parameter.

Reply:
Ok